galaxychop/galaxychop.py

```diff
@@ -107,3 +106,84 @@ def energy(self, pot_star, pot_dark, pot_gas):
 
         return E_tot_star, E_tot_dark, E_tot_gas
 
+    def angular_momentum(self):
+        """Components of the angular momentum of stars, gas and DM.
+
+        Parameters
+        ----------
+
+        Coordinates and velocities for stars: x_s, y_s, z_s and
+         vx_s, vy_s, vz_s.
+
+        Coordinates and velocities for dark matter: x_dm, y_dm, z_dm and
+         vx_dm, vy_dm, vz_dm.
+
+        Coordinates and velocities for gas: x_g, y_g, z_g and vx_g, vy_g, vz_g.
+
+        masses of the stars: m_s.
+
+        Returns
+        -------
+
+        J_part = np.array(3, 3*n), concatenates angular momentum for the stars,
+        dark matter and gas in a single array.
+
+        Jr_star = np.array(n, 1), Component in the plane of the angular
+        momentum of stars.
+
+        Jr = np.array(3*n, 1) Component in the plane of the angular
+        momentum for all particles.
+
+        """
+
+        self.x_s = self.x_s - self.x_dm[p_dm.index(max(p_dm))]
+        self.y_s = self.y_s - self.y_dm[p_dm.index(max(p_dm))]
+        self.z_s = self.z_s - self.z_dm[p_dm.index(max(p_dm))]
+
+        self.x_dm = self.x_dm - self.x_dm[p_dm.index(max(p_dm))]
+        self.y_dm = self.y_dm - self.y_dm[p_dm.index(max(p_dm))]
+        self.z_dm = self.z_dm - self.z_dm[p_dm.index(max(p_dm))]
+
+        self.x_g = self.x_g - self.x_dm[p_dm.index(max(p_dm))]
+        self.y_g = self.y_g - self.y_dm[p_dm.index(max(p_dm))]
+        self.z_g = self.z_g - self.z_dm[p_dm.index(max(p_dm))]
+
+        pos_star_rot, vel_star_rot, A = aling(self.m_s,
+                                              [self.x_s, self.y_s, self.z_s],
+                                              [self.vx, self.vy, self.vz],
+                                              3 .R[j])
+
+        pos_dark_rot = rot([self.x_dm, self.y_dm, self.z_dm], A)
+        vel_dark_rot = rot([self.vx_dm, self.vy_dm, self.vz], A)
+
+        pos_gas_rot = rot([self.x_g, self.y_g, self.z_g], A)
+        vel_gas_rot = rot([self.vx_g, self.vy_g, self.vz_g], A)
```

Comment:
Here you have to be careful, I don't know if you have checked that this works using the files that are in legacy. But at first sight, it seems to me that the vectors (nx3) of pos and vel should be built for each type of particle or the other thing is to change aling and instead of entering vectors of (nx3), they are (nx1). This last one I do not know if I would do it because we already have a test of that function.